### PR TITLE
ci: add spell check with typos

### PR DESCRIPTION
**Pull request by dongxuwang**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

- add spell check with typos in ci workflow

## Checklist

- []  I have written the necessary rustdoc comments.
- []  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
(#589)

### CODE REVIEW

File: typos.toml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[files]
+extend-exclude = ["scripts/*.sh"]
```
Comment (zyy17) [inline typos.toml:2]: Why excludes `scripts/*.sh`?

Comment (dongxuwang) [reply typos.toml:2]: It's checking `set -ue` as a typo.

Comment (waynexia) [reply typos.toml:2]: Maybe maintain an allow-list is a better way https://github.com/marketplace/actions/typos-action#false-positives

Comment (dongxuwang) [reply typos.toml]: Yes, use `default.extend-words` instead of whole file.

**Review by waynexia — APPROVED** [review]
Thanks for this :heart:

File: typos.toml (hunk 1 above)
Comment (zyy17) [reply typos.toml]: @dongxuwang Can we use `.typos.toml`? I think the hidden files will be better?

**Review by evenyag — APPROVED** [review]
LGTM

**Pull request merged**